src/Autocomplete/src/Form/AutocompleteChoiceTypeExtension.php

```diff
@@ -54,6 +54,8 @@ public function finishView(FormView $view, FormInterface $form, array $options):
 
         $attr = $view->vars['attr'] ?? [];
 
+        $attr['required'] = $view->vars['required'];
```

Comment:
Do we still need this? What problem is this solving?

Reply:
tested without - i don't think we do.

Reply:
i see what it's for. it's to make my functional test work:
```php
->assertElementAttributeContains('#product_portionSize', 'required', 'required')
```

this won't work without my change as as symfony does not always render the required form view variable

the javascript will still work though.
up to you...?

Reply:
My thinking is:

A) Symfony very purposely avoids the `required` attribute in illegal situations. It's super important to *not* change that
B) And so, if Symfony moves `required` for one situation, it just means that the clear button will show up in that one edge-case situation... and if I'm reading the code correctly, it already shows up in that situation today.